### DISCUSSION TITLE
HTTPStatusError: Client error '400' for url

### DISCUSSION THREAD

**Discussion by jhardy-lulu** [thoughtspot/cs_tools#101] [General] [answered]
Hey Nick, long time no speak! 

HTTPStatusError: Client error '400 ' for url '[https://lululemon.thoughtspot.cloud/callosum/v1/tspublic/v1/session/login'](https://lululemon.thoughtspot.cloud/callosum/v1/tspublic/v1/session/login%27)
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

I thought it might be because I set config host to https > http

but then i get a entirely diff error when i change to http:

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Comment by boonhapus** [comment]
Hey @jhardy-lulu ! 👋 😄 I just did a check against your cluster from my old configuration and it still works fine. You might try making sure your tools are up to date, and that you're using a `username + password` combination for a User that was created in ThoughtSpot (as opposed to Microsoft AzureAD / SAML)

This won't be a limitation for much longer.. as CS Tools v1.5.0 will offer more ways to authenticate.

1/ Trusted Authentication
Generate a `secret key` which allows you to log in as any user. Available only to ThoughtSpot administrators via the Developer tab in the web interface.

2/ Bearer Token authentication
Generate a long-lived `token`, scoped to a single user.

This should give more flexibility for customers who want to turn off the local authentication option in ThoughtSpot or use their own user.

You can read about them both in the **[V2.0 API Documentation](https://developers.thoughtspot.com/docs/api-authv2)**.

**Reply by jhardy-lulu** [reply, marked as answer]
Wow so you're telling me I just entered my user/pw in wrong? lol
![image](https://github.com/thoughtspot/cs_tools/assets/82538644/9ad8b034-ada2-42bc-ae68-31fdff978d70)
Yep ahah thanks!
